near-sdk/src/collections/legacy_tree_map.rs

```diff
@@ -585,6 +585,16 @@ where
         self.save(&n);
         self.tree.pop();
     }
+
+    pub fn to_tree_map<S: IntoStorageKey>(self, prefix: S) -> super::TreeMap<K, V> {
+        // We read this into the heap first so one can use the same prefix without getting into trouble
+        let elements: Vec<(K, V)> = self.into_iter().collect();
+        let mut tm = super::TreeMap::new(prefix);
+        for (k,v) in elements.iter() {
```

Comment:
This all happens synchronously, what would be a case that this would solve a race condition that using the iterator directly in the loop wouldn't?